drivers/wasapi/audio_driver_wasapi.cpp

```diff
@@ -850,6 +850,27 @@ void AudioDriverWASAPI::thread_func(void *p_udata) {
 								r = read_sample(ad->audio_input.format_tag, ad->audio_input.bits_per_sample, data, j * 2 + 1);
 							} else if (ad->audio_input.channels == 1) {
 								l = r = read_sample(ad->audio_input.format_tag, ad->audio_input.bits_per_sample, data, j);
+							} else if (ad->audio_input.channels > 2) {
+								l = r = 0;
+								int channels = ad->audio_input.channels;
+								for (int ch = 0; ch < channels - 1; ch++) {
+									int32_t sample = read_sample(ad->audio_input.format_tag, ad->audio_input.bits_per_sample, data, j * channels + ch);
+									if (ch % 2 == 0) {
+										r += sample;
+									} else {
+										l += sample;
+									}
+								}
+								int32_t last_sample = read_sample(ad->audio_input.format_tag, ad->audio_input.bits_per_sample, data, j * channels + (channels - 1));
```

Comment:
Can input have the odd number of channels at all (the assumption here seems to be L,R,Center channel config)?
Also, do this odd/even channel distribution to left/right always holds (is defined somewhere in the WASAPI documented, or based on specific mic sample)?

Reply:
I made the assumption that the microphone might have a potential configuration with odd channels (e.g., Left, Right, Center), although such configurations are uncommon (in some professional audio equipment). The goal was to ensure that the microphone is at least usable within Godot, regardless of the channel count. The WASAPI documentation itself doesn't explicitly specify how multi-channel microphones should be handled, so I aimed for a more general solution to accommodate various configurations.